### ISSUE TITLE
Add helper method to get best HyperparameterTuningJob trial

### ISSUE THREAD

**Issue by zchenyu** [googleapis/python-aiplatform#1545]
Currently there's no way to get the best trial from a HyperparameterTuningJob besides inspecting each individual trial and implementing minimization or maximization (depending on the study config).

It would be nice if we had a helper method like HPTuningJob.best_trial() or HPTuningJob.best_trials(n)

Or just sorting the trials returned by trials() might be sufficient.

**Comment by jaycee-li** [comment]
Hi @zchenyu ,

There could be multiple metrics for each trial, so the helper method needs several more fields E.g,`HPTuningJob.best_trials(num_trials=10, metric_id='accuracy', goal='maximize')` 

An alternative approach is to add a `HPTuningJob.get_trails_df()` method which returns all the trials' ids, training steps, params, and metrics in a pandas DataFrame (like what we show in [cloud console UI](https://pantheon.corp.google.com/vertex-ai/locations/us-central1/training/2765355306740350976/cpu?project=sashaproject-1). In this case users can easily sort/filter by every param or metric.

Which option do your prefer?
